### DISCUSSION TITLE
Traffic avoidance

### DISCUSSION THREAD

**Discussion by realm174** [Courseplay/courseplay#6801] [Q&A] [answered]
Is there a way to disable the swerving to avoid traffic? I'm using overloaders and a semi. When the semi goes to the farm to unload, when it returns to its waiting point, if there's already an overloader waiting, the truck will deviate from its designed course to steer further away from the overloader. The problem then is that the semi is too far and the overloader won't unload.  

If the semi is there first, the overloaders do not swerve in that similar manner.

**Comment by pvaiko** [comment]
It would be great if you had a video of this. What mode are you using for the semi? Would rearranging your course a little bit so there's a straight section before the semi reaches the overload point help?

**Reply by realm174** [reply]
For sure. I re-created the situation. See notes in video info section. Hope this helps. Sadly, I marked the video as "for kids" which disabled the comments section, and I have no clue how to change that. 

https://youtu.be/XxcY5D2Ulsw

**Reply by pvaiko** [reply]
@Tensuko didn't we have this problem already?

**Reply by Tensuko** [reply]
We had this when the driver arrives at the waiting point for the overloader only I think.
Disable swerve at overloading point, but I don't think this took Mode1 driver load at start into account.
So we would have to disable swerve near the 1st Waypoint too.

**Reply by pvaiko** [reply]
Created issue #6814 to track this.

**Reply by realm174** [reply]
Thanks for the prompt response. That fixed it.  But I seem to have a different issue now, and I'm not sure if it is related or not.  If you want me to start a new thread, let me know. Please see this video:

https://youtu.be/0ug77h79EPw

As you can see the overloader seem to line up properly by the semi, but at the last minute steers right into it.

**Reply by Tensuko** [reply]
Waitpoint to overload is to far away.
We do not swerver near the Waitpoint, can you move it more to the front of your truck ? I would also delete some Waypoints to the way of the Waitpoint.

**Reply by realm174** [reply]
Thanks Tensuko.  I was also experimenting since I posted this and did fix it.  I moved the semi further back (so I would have more room to navigate the overloader) and made the overloader's course shorter and in a more or less straight line, with the Waitpoint at the front edge of the trailer. That seems to fix it.  This field is gonna be about 30 loads or so for the overloader, so I'll keep an eye and report back if I have issues. For the time being, let's consider this fixed. Thanks again for the prompt response!

**Reply by pvaiko** [reply]
> We do not swerve near the Waitpoint, can you move it more to the front of your truck ? I would also delete some Waypoints to the way of the Waitpoint.

We disable swerve within 30 meters of the wait point.

**Comment by pvaiko** [comment, marked as answer]
#6814
